cogboard-webapp/src/actions/thunks.js

```diff
@@ -65,12 +65,23 @@ export const login = credentials => dispatch => {
   );
 };
 
-export const logout = () => dispatch => {
+export const logout = () => (dispatch, getState) => {
   const userRole = getUserRole();
+  const {
+    app: { logoutReasonMessage }
+  } = getState();
 
   removeToken();
   dispatch(logoutUser());
-  dispatch(pushNotification(NOTIFICATIONS.LOGOUT(userRole)));
+  if (logoutReasonMessage !== '') {
+    dispatch(
+      pushNotification(
+        NOTIFICATIONS.LOGOUT_REASON(userRole, logoutReasonMessage)
```

Comment:
Using common function for this logic would also simplify this block of code to just set logoutReasonMessage and dispatch action at the end.